README
Dist::Zilla::Plugin::Author::KENTNL::CONTRIBUTING

This is a personal Dist::Zilla plug-in that generates a CONTRIBUTING
section in my distribution.

I would have made something more general, but my head exploded in
thinking about it.

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-Plugin-Author-KENTNL-CONTRIBUTING

0.001007 2017-03-04T01:38:18Z 3e3671d

0.001006 2017-03-04T01:21:44Z 48557a0
 [Bugfix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Plugin-Author-KENTNL-CONTRIBUTING/issues/2

 [Dependencies::Stats]
 - Dependencies changed since 0.001005, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑1)
 - runtime: -1
 - test: ↓1

 [Internals]
 - Drop ConfigDumper

0.001005 2015-11-14T04:44:40Z 09c7edb
 - switched from [GenerateFile::ShareDir] to [GenerateFile::FromShareDir], closes #1

 [Dependencies::Stats]
 - Dependencies changed since 0.001004, see misc/*.deps* for details
 - develop: +3 ↑1 -1 (suggests: ↑1)
 - runtime: +1 -3 (recommends: -1)

0.001004 2015-07-05T06:07:52Z 0f96d87
 [Dependencies::Stats]
 - Dependencies changed since 0.001003, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: ↑1 -2

 [Features]
 - Reimplemented some remaning parts to just use new features in GenerateFile::ShareDir 0.006

 [Todo]
 - Eliminate need for this module at some point without needing my bundle also somehow with some intermediary magic
   that will just call GF:SD outside the normal dzil things(!??)

0.001003 2015-06-30T05:57:29Z 56a04ee
 [Dependencies::Stats]
 - Dependencies changed since 0.001002, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)
 - runtime: +2 -3

 [Features]
 - Reimplement in terms of GenerateFile::ShareDir

0.001002 2015-06-30T02:08:26Z 473269e
 [Bugfix]
 - Turns out to be useful to actually install files intended for a sharedir

 [Dependencies::Stats]
 - Dependencies changed since 0.001001, see misc/*.deps* for details
 - configure: +1 (recommends: +1)
 - develop: +1

0.001001 2015-06-30T00:24:44Z 24a9a5d
 [Content]
 - Credit original author in emitted file.

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - develop: +1 -1

0.001000 2015-06-29T23:35:19Z 9e726b2
 - First version.


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (qw( KENTNL ));
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

#$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');
#$bundle->remove_policy('CodeLayout::RequireUseUTF8');
#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Generates a CONTRIBUTING file for KENTNL's distributions.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Author-KENTNL-CONTRIBUTING",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Author::KENTNL::CONTRIBUTING",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::GenerateFile::FromShareDir" => "0.006",
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.001007",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Plugin::GenerateFile::FromShareDir" => "0.006",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Moose" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
